### PR TITLE
Implement Custom Dropdown

### LINKED ISSUE #28: Location select drop-down doesn't work as expected on mobile
I see that the suggestions come up on Android keyboard as text suggestions but I think that might be a bit confusing for users. I wonder if we can change this back to a webpage drop-down. How does it show up on iOS?

### PR DESCRIPTION
**Pull request by ddamato**
Fixes https://github.com/stanchanpsu/whencanipickle/issues/28 (_hopefully_).

This replaces the incomplete behavior of the native `<datalist/>` in favor of a more feature-complete experience:
- `<input/>` filters the list of locations.
- Selecting a location will reference a specific entry in the data instead of doing a fuzzy match with the label.
- Flyout opens/closes at appropriate times using `aria-expanded`.
- Keyboard navigation with ArrowUp/ArrowDown, handling tabIndex.
- Keep the highlighted element visible while traversing.

Also other small fixes:
- Hides the map `<iframe/>` until given a `src`.
- Smooth scrolling behavior on the page.

### CODE REVIEW

**Comment by stanchanpsu** [comment]
Seems to work on android :) I'll review.

File: src/layouts/Layout.astro [hunk 1]
```diff
@@ -22,4 +22,8 @@ import Body from '@components/Body.astro';
         --foreground-color: light-dark(#222222, #dddddd);
         --background-color: light-dark(#ffffff, #333333);
     }
+
+    html {
+        scroll-behavior: smooth;
```
Comment (stanchanpsu) [inline src/layouts/Layout.astro:27]: why is this not the default lol

File: src/scripts/location-select.js [hunk 2]
```diff
@@ -24,8 +26,7 @@ function formatLabel(entry) {
  * @returns {Boolean}
  */
 function isFuture(startTime) {
-    const startTimestamp = new Date(startTime);
-    return startTimestamp >= new Date()
+    return new Date(startTime) >= new Date()
```
Comment (stanchanpsu) [inline src/scripts/location-select.js:29]: 👍

File: src/scripts/location-select.js [hunk 3]
```diff
@@ -131,6 +210,26 @@ fetch('/cities.json').then((res) => res.json()).then((cities) => {
...
-    $input?.addEventListener('change', onChange);
-    onChange();
-});
\ No newline at end of file
+    /**
+     * Triggers several functions when a location is selected.
+     * 
+     * @param {Element} $btn - The element selected by the user
+     */
+    function userChosen($btn = $locations.querySelector('button[tabIndex="0"]')) {
```
Comment (stanchanpsu) [inline src/scripts/location-select.js:218]: nit: `locationSelected`?

File: src/scripts/location-select.js [hunk 4]
```diff
@@ -99,27 +100,105 @@ function goodForecasts(acc, forecast) {
...
+    if (!$current) return;
+    [...$locations.children].forEach(($btn) => {
+        $btn.tabIndex = $btn === $current ? 0 : -1;
+    });
+    $current.scrollIntoView({ block: 'nearest' });
+    return $current;
+}
+
+// Show or hide the flyout depending on if the input has a value.
+$input.addEventListener('click', (ev) =>
```
Comment (stanchanpsu) [inline src/scripts/location-select.js:136]: it's funny to see all these top level event listeners being added back in lol

**Review by stanchanpsu — APPROVED** [review]
TYVM!

File: src/scripts/location-select.js (hunk 3 above)
Comment (ddamato) [reply src/scripts/location-select.js:218]: I'm being specific about when the user chooses the location versus when the system does. I could align the function names to reflect that.

File: src/scripts/location-select.js (hunk 4 above)
Comment (ddamato) [reply src/scripts/location-select.js:136]: Yeah, they're necessary again since `<datalist/>` isn't what I expected.

**Pull request merged**